go/arrow/compute/internal/kernels/constant_factor_amd64.go

```diff
@@ -0,0 +1,17 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kernels
```

Comment:
Am I missing something, or is there supposed to be associated (generated?) code to go along with this?

Reply:
Well, you see. I'm just an idiot who scaffolded everything out to do the generated SIMD versions of the code and then got sidetracked by everything else and completely forgot leaving it to just run with the native pure go versions lol. 

Let me fix that and update this PR. Thanks!

Reply:
Ah, I was wondering if Go supported generating architecture-specific code based on particular naming conventions or something :sweat_smile:

Reply:
Lol, well the naming convention does control files which are compiled in for specific architectures. For example a file that ends with `_amd64.go` will only be compiled if compiling for x86_64 and a file ending with `_arm64.go` will only get compiled with building for an arm64 cpu